estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from datetime import timedelta
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Estate Property'
+
+    price = fields.Float(string='Price')
+    status = fields.Selection(
+        string='Status',
+        selection=[('accepted', 'Accepted'), ('refused', 'Refused')],
+        copy=False
+    )
+    partner_id = fields.Many2one('res.partner', required=True)
+    property_id = fields.Many2one(
+        'estate.property', required=True, ondelete='cascade')
+    create_date = fields.Date()
+    property_type_id = fields.Many2one(
+        comodel_name='estate.property.type',
+        related='property_id.property_type_id',
+        store=True
+    )
+    validity = fields.Integer(string='Validity', default=7)
+    date_deadline = fields.Date(
+        string='Deadline', compute='_compute_deadline', inverse='_inverse_deadline')
+
+    _sql_constraints = [
+        ('check_offer_price', 'CHECK(price > 0)',
+         'The offer price must be strictly positive.'),
+    ]
+
+    @api.depends('create_date', 'validity')
+    def _compute_deadline(self):
+        for offer in self:
+            if offer.create_date:
+                offer.date_deadline = offer.create_date + \
+                    timedelta(days=offer.validity)
+            else:
+                offer.date_deadline = False
+
+    def _inverse_deadline(self):
+        for offer in self:
+            if offer.create_date and offer.date_deadline:
+                dayys = (offer.date_deadline - offer.create_date).days
+                offer.validity = dayys
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for record in vals_list:
+            property_id = record.get('property_id')
+            if property_id:
+                existing_offers = self.search(
+                    [('property_id', '=', property_id)])
+                if existing_offers and any(offer.price >= vals_list[0]['price'] for offer in existing_offers):
+                    raise UserError(
+                        'You cannot create an offer lower than or equal to existing offer.')
```

Comment:
```suggestion
        if property_id:
            existing_offer = self.search([('property_id', '=', property_id)], limit=1, order='price DESC')
            if existing_offer and existing_offer.price >= price:
                raise UserError('You cannot create an offer lower than or equal to an existing offer.')

```

can we do like this for avoids unnecessary loops.